service/matching/physical_task_queue_manager.go

```diff
@@ -213,27 +212,27 @@ func newPhysicalTaskQueueManager(
 ) (*physicalTaskQueueManagerImpl, error) {
 	e := partitionMgr.engine
 	config := partitionMgr.config
-	logger := log.With(partitionMgr.logger, tag.WorkerBuildId(queue.VersionSet()))
-	throttledLogger := log.With(partitionMgr.throttledLogger, tag.WorkerBuildId(queue.VersionSet()))
 	buildIdTagValue := queue.VersionSet()
 	if buildIdTagValue == "" {
 		buildIdTagValue = queue.BuildId()
 	}
-	taggedMetricsHandler := partitionMgr.taggedMetricsHandler.WithTags(
+	logger := log.With(partitionMgr.logger, tag.WorkerBuildId(buildIdTagValue))
+	throttledLogger := log.With(partitionMgr.throttledLogger, tag.WorkerBuildId(buildIdTagValue))
+	taggedMetricsHandler := partitionMgr.metricsHandler.WithTags(
 		metrics.OperationTag(metrics.MatchingTaskQueueMgrScope),
-		metrics.WorkerBuildIdTag(buildIdTagValue))
+		metrics.WorkerBuildIdTag(buildIdTagValue, config.BreakdownMetricsByBuildID()))
```

Comment:
dynamic config change won't take effect until task queue is reloaded.. I guess that's okay

Reply:
yes it should be fine.